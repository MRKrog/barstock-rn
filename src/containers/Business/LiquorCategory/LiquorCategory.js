import React, { Component } from "react";
import styles from './LiquorCategory.style';
import { TouchableOpacity, ScrollView, Text, StyleSheet, Dimensions, View, LayoutAnimation } from "react-native";
import { connect } from "react-redux";
import * as actions from "../../../redux/actions";

export class LiquorCategory extends Component{
  constructor(){
    super();
    this.state ={

    }
  }

  changeLayout = () => {
    LayoutAnimation.configureNext(LayoutAnimation.create(100));
    this.setState({ expanded: !this.state.expanded });
  }

  changeTypeLayout = (type) => {
    this.props.changeType(type)
    this.setState({ expanded: !this.state.expanded });
  }

  render(){
    const alcoholTypes = this.props.allTypes.map(type => {
      return (
        <TouchableOpacity onPress={() => this.changeTypeLayout(type)} key={type}>
          <Text style={styles.alc_types}>{type.toUpperCase()}</Text>
        </TouchableOpacity>
      )
    })

    return (
      <View >
        <TouchableOpacity onPress={this.changeLayout}>
          <Text style={styles.alc_typeTitle}>
            {this.props.currentType}
          </Text>
        </TouchableOpacity>
        <ScrollView>
          <View style={[{ height: this.state.expanded ? null : 0, overflow: 'hidden' }]}>
            { alcoholTypes }
          </View>
        </ScrollView>
      </View>
    )
  }
}

export const mapStateToProps = (state) => ({
  cart: state.cart
})

export const mapDispatchToProps = (dispatch) => ({
  addToCart: alcohol => dispatch(actions.addToCart(alcohol)),
  removeFromCart: alcohol => dispatch(actions.removeFromCart(alcohol))
})

export default connect(mapStateToProps, mapDispatchToProps)(LiquorCategory)
