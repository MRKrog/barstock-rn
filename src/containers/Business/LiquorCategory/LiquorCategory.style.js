export default {
  container: {
    display: "flex",
  },
  alc_typeTitle: {
    fontSize: 23,
    textTransform: "uppercase",
    color: "#ffffff",
    textAlign: "center",
    fontFamily: 'abel',
    display: "flex"
  },
  expandedFalse: {
    marginTop: 0
  },
  expandedTrue: {
    marginTop: 10,
    backgroundColor: "#11212A",
    borderRadius: 5,
  },
  alc_typeContainer: {
    borderRadius: 5
  },
  alc_types: {
    fontSize: 20,
    lineHeight: 36,
    textTransform: "uppercase",
    color: "#ffffff",
    textAlign: "center",
    fontFamily: 'abel',
  },
  iconStyle: {
    position: "absolute",
    right: 2,
    fontSize: 25,
    color: "#ffffff"
  },
  alc_toggle: {
    color: "#ffffff",
    fontSize: 30,
    right: 5,
    top: -5,
    position: "absolute"
  }
}
