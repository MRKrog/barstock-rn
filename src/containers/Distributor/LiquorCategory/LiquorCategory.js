import React, { Component } from "react";
import styles from './LiquorCategory.style';
import { TouchableOpacity, ScrollView, Text, View, LayoutAnimation } from "react-native";
import { connect } from "react-redux";
import * as actions from "../../../redux/actions";
import { Icon } from 'native-base';

export class LiquorCategory extends Component{
  constructor(){
    super();
    this.state = {
      expanded: false
    }
  }
  changeLayout = () => {
    LayoutAnimation.configureNext(LayoutAnimation.create(100));
    this.setState({ expanded: !this.state.expanded });
  }

  changeTypeLayout = (type) => {
    this.props.changeType(type)
    this.setState({ expanded: !this.state.expanded });
  }

  render(){
    const alcoholTypes = this.props.allTypes.map(type => {
      return (
        <TouchableOpacity onPress={() => this.changeTypeLayout(type)} key={type} style={styles.alc_typeContainer}>
          <Text style={styles.alc_types}>
          <Icon name='beer' color='#ffffff' size={20} style={styles.iconStyle} />
          {type.toUpperCase()}
          </Text>
        </TouchableOpacity>
      )
    })

    let expandedStyles;
    if(this.state.expanded === true){
      expandedStyles = styles.expandedTrue
    } else {
      expandedStyles = styles.expandedFalse
    }

    return (
      <View style={styles.container}>
        <TouchableOpacity onPress={this.changeLayout}>
          <Text style={styles.alc_typeTitle}>
            {this.props.currentType}
          </Text>
        </TouchableOpacity>
        <ScrollView style={styles.alc_typeContainer}>
          <View style={[expandedStyles, { height: this.state.expanded ? null : 0, overflow: 'hidden' }]}>
            { alcoholTypes }
          </View>
        </ScrollView>
      </View>
    )
  }
}

export const mapStateToProps = (state) => ({
  cart: state.cart
})

export const mapDispatchToProps = (dispatch) => ({
  addToCart: alcohol => dispatch(actions.addToCart(alcohol)),
  removeFromCart: alcohol => dispatch(actions.removeFromCart(alcohol))
})

export default connect(mapStateToProps, mapDispatchToProps)(LiquorCategory)
