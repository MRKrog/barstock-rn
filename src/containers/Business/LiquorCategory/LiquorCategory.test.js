import { shallow } from "enzyme"
import { LiquorCategory } from "./LiquorCategory"
import React from "react"

describe("LiquorCategory", () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <LiquorCategory />
    )
  })

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot()
  })
})
