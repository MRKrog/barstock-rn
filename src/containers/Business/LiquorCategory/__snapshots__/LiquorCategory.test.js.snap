// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiquorCategory should match snapshot 1`] = `undefined`;
